src/chat.rs

```diff
@@ -928,6 +931,42 @@ impl Chat {
     }
 }
 
+#[derive(Debug, Copy, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub enum ChatVisibility {
+    Normal = 0,
+    Archived = 1,
+    Pinned = 2,
+}
+
+impl rusqlite::types::ToSql for ChatVisibility {
+    fn to_sql(&self) -> rusqlite::Result<rusqlite::types::ToSqlOutput> {
+        let duration = match &self {
+            ChatVisibility::Normal => 0,
+            ChatVisibility::Archived => 1,
+            ChatVisibility::Pinned => 2,
+        };
+        let val = rusqlite::types::Value::Integer(duration as i64);
+        let out = rusqlite::types::ToSqlOutput::Owned(val);
+        Ok(out)
+    }
+}
+
+impl rusqlite::types::FromSql for ChatVisibility {
+    fn column_result(value: rusqlite::types::ValueRef) -> rusqlite::types::FromSqlResult<Self> {
+        i64::column_result(value).and_then(|val| {
+            match val {
+                2 => Ok(ChatVisibility::Pinned),
+                1 => Ok(ChatVisibility::Archived),
+                0 => Ok(ChatVisibility::Normal),
+                n => {
+                    // unknown archived state, falling back to normal state (was this db opened with a newer deltachat version?)
```

Comment:
This comment is wrong, nothing here falls back but an error is returned instead.  I think the code is right in doing this and that the comment should be removed.  Though I suspect @r10s might take the view that the comment is right and the code should change.  Most existing FromSql impls return an error here I think.

Reply:
good catch. and yes, i would probably prefer to read the record even if the state is known, otherwise a whole chat will be gone permantently.